File: python/Sub-array elements sum.py
# https://www.codewars.com/kata/5b5e0ef007a26632c400002a/train/python

def elements_sum(arr, d):
    result = 0
    n = len(arr)
    for subarr in range(n):
        index = n - subarr - 1
        if len(arr[subarr]) > index:
            result += arr[subarr][index]
        else:
            result += d
    return result

File: Cpp/Sub-array elements sum.cpp
#include<vector>
using namespace std;

long elementsSum(const vector<vector<int>>& arr, int d = 0) {
    long result = 0;
    int arrSize = arr.size();

    if (arrSize == 0) 
    {
        return result;
    }

    for(int subArr = 0; subArr < arrSize; subArr++) 
    {
        int index = (arrSize - 1) - subArr;
        
        if (index < arr[subArr].size()) 
        {
            result += arr[subArr][index];
        }
        else 
        {
            result += d;
        }
    }

    return result;
}
